### ISSUE TITLE
Appreciation & Code Request for Comparison Algorithms

### ISSUE THREAD

**Issue by wenxinyang1026** [XeniaLLL/FOOGD-main#1]
Thanks for sharing the codebase! Your work in src.algorithms has greatly accelerated my research.
Would be awesome if you could also share implementations like ​​FedICON​​/​​FedTHE​​ for benchmarking.

**Comment by XeniaLLL** [comment]
Yes, you can refer to it. But it would be better if you highlight the source and cite the related publications.

**Comment by wenxinyang1026** [comment]
Thank you, but I still can't find the implementations like ​​FedICON​​/​​FedTHE in src/algorithms. Hope you can share them, thank you very much!

**Comment by XeniaLLL** [comment]
You can easily re-implement them from the source of https://github.com/LINs-lab/FedTHE.git and https://openreview.net/forum?id=qJJmu4qsLO. This project is mainly about sharing the implementation of FOOGD.

**Comment by wenxinyang1026** [comment]
Thank you for your suggestions. I have reproduced the algorithm FedICON, keeping the hyperparameters consistent with those in the paper, and used the publicly available implementations of SimCLR and SupContrast for contrastive learning loss. However, the test performance decreased as the number of training rounds increases. I understand that this project is mainly about sharing the implementation of FOOGD. I’m sorry to bother you, but I would be truly grateful if you could share your reproduced FedICON code with me via email at 2174089673@qq.com. Thank you so much in advance.
